openedx/core/djangoapps/monitoring_utils/__init__.py

```diff
@@ -110,3 +111,14 @@ def function_trace(function_name):
             yield
     else:
         yield
+
+
+def graph_memory_leaks(request):
+    """
+    Run anywhere inside a view to have the memory monitoring middleware create
+    diagnostic memory reference graphs for new objects leaked while executing
+    it.  That middleware must be enabled via a waffle switch for this to have
+    any effect.
+    """
+    view_name = inspect.stack()[1][3]
+    setattr(request, middleware.MonitoringMemoryMiddleware.memory_graphs_key, view_name)
```

Comment:
Nit: It feels a bit weird for this method to know `MonitoringMemoryMiddleware.memory_graphs_key` is the right field to stick into `request`, which will then be picked up on the other end. Is it possible to call a class method on the `MonitoringMemoryMiddleware` that would pull down the current request and get the data it needs from it?

Also, where is this method called?